l10n_br_account/models/account_invoice.py

```diff
@@ -316,6 +316,94 @@ def onchange_fiscal_document_id(self):
         if self.issuer == '0':
             self.document_serie_id = self.company_id.document_serie_service_id
 
+    @api.multi
+    def invoice_validate(self):
+        super(AccountInvoice, self).invoice_validate()
+        for object in self.env['account.invoice'].search(
+                [('id', '=', self.id)]):
+            if (not object.journal_id.revenue_expense
+                and object.journal_id.automatic_conciliation
+                and object.fiscal_category_id.property_journal ==
+                    object.journal_id
+                and object.state == 'open'
+                and object.company_id.id ==
+                    self.env.user.company_id.id
+                and object.journal_id.conciliation_journal):
+                voucher_obj = self.env['account.voucher']
```

Comment:
Yes. It's better if we can do this in another module.

Luís Felipe Mileo
________________________________
De: Danimar Ribeiro <notifications@github.com>
Enviado: 16/06/2016 12:10 PM
Para: OCA/l10n-brazil
Cc: Luis Felipe Miléo; Comment
Assunto Re: [OCA/l10n-brazil] Herda método invoice_validate para conciliar automaticamente faturas … (#333)

> @@ -316,6 +316,94 @@ def onchange_fiscal_document_id(self):
>          if self.issuer == '0':
>              self.document_serie_id = self.company_id.document_serie_service_id
>  
> +    @api.multi
> +    def invoice_validate(self):
> +        super(AccountInvoice, self).invoice_validate()
> +        for object in self.env['account.invoice'].search(
> +                [('id', '=', self.id)]):
> +            if (not object.journal_id.revenue_expense
> +                and object.journal_id.automatic_conciliation
> +                and object.fiscal_category_id.property_journal ==
> +                    object.journal_id
> +                and object.state == 'open'
> +                and object.company_id.id ==
> +                    self.env.user.company_id.id
> +                and object.journal_id.conciliation_journal):
> +                voucher_obj = self.env['account.voucher']

Isn't necessary account_voucher in the dependency?

---
You are receiving this because you commented.
Reply to this email directly or view it on GitHub:
https://github.com/OCA/l10n-brazil/pull/333/files/f374ba495a07b0b0ee325b032c60d447d9318b55#r67363307